CIP-0072/README.md

```diff
@@ -0,0 +1,560 @@
+The consequence of this will be that as we pass different parameters, script hash will be changing. This is especially troublesome for things like certifications / audits but also dApp registration. This topic is being debated as part of CIP: https://github.com/cardano-foundation/CIPs/pull/385, however, it doesn't see that there has been conclusion how to tackle this problem. For the moment, a new script hash (despite changing only a parameter) requires a re REGISTRATION or an UPDATE to the existing dApp.
+
+### Often Changing Scripts
+There are cases on Cardano main-net that script hashes are changing every day, most probably due to parameterised scripts. It is responsibility of the developers to issue an `UPDATE` command and provide on-chain and off-chain metadata following the change, for scripts that are changing daily / hourly it is envisaged that this process be automated by a dApp developer.
+
+### Beacon Tokens Instead of Metadata
+It has been argued that since anybody can make claims to dApps, this CIP instead of using metadata should use tokens. dApp developers
+would mint token, which would ensure they are the owners of a given dApp. It is a certainly an interesting approach but shortcomings 
+of the current solution can also be lifted by moving to DID based system and benefit of metadata is that it is easily queriable off chain
+and currently stores can attest / validate multiple claims for the same dApp. Forcing dApp developers to MINT tokens would complicate this CIP
+and potentially hinder it's adoption.
+
+### Datums Instead of Metadata 
+It has been suggested that we do not use metadata but rather Datums. Metadata cannot enforce format and Datums could. It has been rejected as
+using Datums requires a smart contract and we want to keep this solution as accessible as possible. It is a GUI concern since if there is a 
+some app that can attest validity and conformance to JSON schema - dApp Registration / Update MUST never be done that does not conform to the schema.
+
+### Scripts / Releases Fields Are Not Required
+We made a decision to change the schema so that scripts and releases are no longer required. This could help to get initial registration from dApp developers faster and
+some stores simply do not require dApps to add their scripts in order to be listed.
+
+### Schema Version or CIP Version
+We want to take schema version further and actually attest the whole CIP version so we are introducing a top level field: version. Any changes to the document will require bumping the version. CIP version needs to be also
+propagated in the on-chain JSON itself.
+
+### Tags
+We briefly discussed tags and we will likely introduce tags in the near future. An array of tags to help stores / dApp developers categories where their dApp should show. This will complement `categories` field.
+
+### DE-REGISTER
```

Comment:
This addition raises a question for me:

- What should happen if a publisher registers then de-registers then registers again? is this a valid flow?